### ISSUE TITLE
Does some update of paper recently break PaperMake?

### ISSUE THREAD

**Issue by qyl27** [Rikonardo/PaperMake#10]
I have used PaperMake for several months, but in my new plugin project, the hot reload of PaperMake is not working.

When I run a build while dev server running, my empty plugin was always shows failed to load.  
In 1.19.2 it works well, but 1.19.3 has this error.  

Stack traces are below:  
```
[13:36:34 INFO]: [PaperMakeHook] Reloading...
[13:36:34 ERROR]: [ModernPluginLoadingStrategy] Could not load plugin 'uhvxg4mGleP8xH49lPa31197nK37WH1W.jar' in folder 'C:\Workspace\AlchemyIO\build\papermake\build\artifacts'
java.lang.IllegalStateException: Provider SpigotPluginProvider{path=C:\Workspace\AlchemyIO\build\papermake\build\artifacts\uhvxg4mGleP8xH49lPa31197nK37WH1W.jar, description=org.bukkit.plugin.PluginDescriptionFile@2bde1403, jarFile=java.util.jar.JarFile@5ee5c233, status=null, dependencyContext=null} attempted to add duplicate plugin identifier AlchemyIO v1.0.0 THIS WILL CREATE BUGS!!!
	at io.papermc.paper.plugin.storage.ServerPluginProviderStorage$1.applyContext(ServerPluginProviderStorage.java:28) ~[paper-1.19.3.jar:git-Paper-426]
	at io.papermc.paper.plugin.entrypoint.strategy.ModernPluginLoadingStrategy.loadProviders(ModernPluginLoadingStrategy.java:147) ~[paper-1.19.3.jar:git-Paper-426]
	at io.papermc.paper.plugin.storage.SimpleProviderStorage.enter(SimpleProviderStorage.java:35) ~[paper-1.19.3.jar:git-Paper-426]
	at io.papermc.paper.plugin.manager.SingularRuntimePluginProviderStorage.enter(SingularRuntimePluginProviderStorage.java:63) ~[paper-1.19.3.jar:git-Paper-426]
	at io.papermc.paper.plugin.manager.RuntimePluginEntrypointHandler.enter(RuntimePluginEntrypointHandler.java:40) ~[paper-1.19.3.jar:git-Paper-426]
	at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.loadPlugin(PaperPluginInstanceManager.java:121) ~[paper-1.19.3.jar:git-Paper-426]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.loadPlugin(PaperPluginManagerImpl.java:82) ~[paper-1.19.3.jar:git-Paper-426]
	at org.bukkit.plugin.SimplePluginManager.loadPlugin(SimplePluginManager.java:406) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
	at com.rikonardo.papermake.hook.utils.PluginUtil.load(PluginUtil.kt:125) ~[_papermake_hook.jar:?]
	at com.rikonardo.papermake.hook.HookManager.load(HookManager.kt:68) ~[_papermake_hook.jar:?]
	at com.rikonardo.papermake.hook.HookManager.reload(HookManager.kt:88) ~[_papermake_hook.jar:?]
	at com.rikonardo.papermake.hook.HookManager$setup$1$2.invoke$lambda-0(HookManager.kt:36) ~[_papermake_hook.jar:?]
	at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.19.3.jar:git-Paper-426]
	at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:483) ~[paper-1.19.3.jar:git-Paper-426]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1473) ~[paper-1.19.3.jar:git-Paper-426]
	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:447) ~[paper-1.19.3.jar:git-Paper-426]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1397) ~[paper-1.19.3.jar:git-Paper-426]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1173) ~[paper-1.19.3.jar:git-Paper-426]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316) ~[paper-1.19.3.jar:git-Paper-426]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
[13:36:34 WARN]: org.bukkit.plugin.InvalidPluginException: Plugin didn't load any plugin providers?
[13:36:34 WARN]: 	at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.lambda$loadPlugin$1(PaperPluginInstanceManager.java:127)
[13:36:34 WARN]: 	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
[13:36:34 WARN]: 	at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.loadPlugin(PaperPluginInstanceManager.java:127)
[13:36:34 WARN]: 	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.loadPlugin(PaperPluginManagerImpl.java:82)
[13:36:34 WARN]: 	at org.bukkit.plugin.SimplePluginManager.loadPlugin(SimplePluginManager.java:406)
[13:36:34 WARN]: 	at _papermake_hook.jar//com.rikonardo.papermake.hook.utils.PluginUtil.load(PluginUtil.kt:125)
[13:36:34 WARN]: 	at _papermake_hook.jar//com.rikonardo.papermake.hook.HookManager.load(HookManager.kt:68)
[13:36:34 WARN]: 	at _papermake_hook.jar//com.rikonardo.papermake.hook.HookManager.reload(HookManager.kt:88)
[13:36:34 WARN]: 	at _papermake_hook.jar//com.rikonardo.papermake.hook.HookManager$setup$1$2.invoke$lambda-0(HookManager.kt:36)
[13:36:34 WARN]: 	at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftTask.run(CraftTask.java:101)
[13:36:34 WARN]: 	at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:483)
[13:36:34 WARN]: 	at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1473)
[13:36:34 WARN]: 	at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:447)
[13:36:34 WARN]: 	at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1397)
[13:36:34 WARN]: 	at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1173)
[13:36:34 WARN]: 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316)
[13:36:34 WARN]: 	at java.base/java.lang.Thread.run(Thread.java:833)
[13:36:34 WARN]: [PaperMakeHook] 1 plugin failed to load, check console for details
[13:36:34 INFO]: [PaperMakeHook] Plugin reloaded!
```

**Comment by Rikonardo** [comment]
Looks like Paper introduced their own plugin loader. I'll work on fix today

**Comment by Rikonardo** [comment]
Fixed in new **[v1.0.6](https://plugins.gradle.org/plugin/com.rikonardo.papermake/1.0.6)** release

**Comment by qyl27** [comment]
Yes, I just tried, it works!  
Thank you a lot.
